tests/phpunit/includes/MediaWiki/Hooks/InfoActionTest.php

```diff
@@ -0,0 +1,338 @@
+<?php
+
+namespace SMW\Tests\MediaWiki\Hooks;
+
+use SMW\Tests\Utils\Mock\MockTitle;
+
+use SMW\MediaWiki\Hooks\InfoAction;
+
+use SMW\Settings;
+use SMW\ApplicationFactory;
+use SMW\DIWikiPage;
+use SMW\DIProperty;
+
+use ParserOutput;
+use Language;
+
+/**
+ * @covers \SMW\MediaWiki\Hooks\InfoAction
+ *
+ *
+ * @group SMW
+ * @group SMWExtension
+ * @group medium
+ *
+ * @license GNU GPL v2+
+ * @since 2.3
+ *
+ * @author misdre
+ */
+class InfoActionTest extends \PHPUnit_Framework_TestCase {
+
+	private $applicationFactory;
+
+	protected function setUp() {
+		parent::setUp();
+
+		$this->applicationFactory = ApplicationFactory::getInstance();
+
+		$settings = Settings::newFromArray( array(
+			'smwgShowFactbox'                 => SMW_FACTBOX_NONEMPTY,
+			'smwgFactboxUseCache'             => true,
+			'smwgCacheType'                   => 'hash',
+			'smwgLinksInValues'               => false,
+			'smwgInlineErrors'                => true,
+		) );
+
+		$this->applicationFactory->registerObject( 'Settings', $settings );
+	}
+
+	protected function tearDown() {
+		$this->applicationFactory->clear();
+
+		parent::tearDown();
+	}
+
+	public function testCanConstruct() {
+
+		$pageInfo = '';
+
+		$outputPage = $this->getMockBuilder( '\OutputPage' )
+			->disableOriginalConstructor()
+			->getMock();
+
+		$parserOutput = $this->getMockBuilder( '\ParserOutput' )
+			->disableOriginalConstructor()
+			->getMock();
+
+		$this->assertInstanceOf(
+			'\SMW\MediaWiki\Hooks\InfoAction',
+			new InfoAction( $outputPage, $parserOutput, $pageInfo )
+		);
+	}
+
+	/**
+	 * @dataProvider outputDataProvider
+	 */
+	public function testProcess( $parameters, $expected ) {
```

Comment:
This test, and it's data provider also seem problematic. Both methods are quite big, and the amount of mocking going on is unfortunate. To me this test is hard to understand, and it certainly has a lot of coupling to internal details of the production code it tests, making it hard to make modifications.